core/src/main/scala/kafka/log/LogLoader.scala

```diff
@@ -368,12 +368,40 @@ object LogLoader extends Logging {
     for (swapFile <- swapFiles) {
       val logFile = new File(CoreUtils.replaceSuffix(swapFile.getPath, Log.SwapFileSuffix, ""))
       val baseOffset = Log.offsetFromFile(logFile)
+
+       def maybeCompleteInterruptedSwap(fn: (File, Long, String) => File): Boolean = {
+        val swapOffsetIndexFile = fn(swapFile.getParentFile, baseOffset, Log.SwapFileSuffix)
+        if (swapOffsetIndexFile.exists())
+          true
+        else {
+          val cleanedOffsetIndexFile = Log.offsetIndexFile(swapFile.getParentFile, baseOffset, Log.CleanedFileSuffix)
+          if (cleanedOffsetIndexFile.exists() && cleanedOffsetIndexFile.renameTo(swapOffsetIndexFile)) true
+          else false
+        }
+      }
+
+      // Check whether swap index files exist: if not, the cleaned files must exist due to the
+      // existence of swap log file. Therefore, we rename the cleaned files to swap files and continue.
+      val recoverable = maybeCompleteInterruptedSwap(Log.offsetIndexFile) &&
```

Comment:
`recoverable` sounds a bit incorrect in this context, given that we actually end up calling `recoverSegment` when `recoverable == false`.

Perhaps we could call this something like `needsRecovery` which is set to `true` if we do not find the relevant index files or when the sanity check fails.